### PR TITLE
unlocking notebook breaks code cells #3080

### PR DESCRIPTION
**Pull request by kpaharelau**
Functionality for the refresh codemirror after unhide cell has been added. In current momemt we are using 2 way for refreshing codemirror:
1. After changing 'IsLocked' property
   2/ After click on "Unhide" menu button

### CODE REVIEW

File: core/src/main/web/app/mainapp/components/notebook/codecell-directive.js [hunk 1]
```diff
@@ -25,7 +45,8 @@
       bkSessionManager,
       bkCoreManager,
       bkPublicationHelper,
-      $timeout) {
+      $timeout,
+      $animate) {
```
Comment (tomca32) [inline core/src/main/web/app/mainapp/components/notebook/codecell-directive.js]: why do we need animate here? It seems that it's used only in ngHideEx directive.

File: core/src/main/web/app/mainapp/components/notebook/codecell-directive.js [hunk 2]
```diff
@@ -70,22 +91,24 @@
...
+
+        //ensure cm refreshes when 'unhide'
+        $scope.afterShow = function () {
+          $scope.cm.refresh();
+        };
+        $scope.$watch('cellmodel.input.hidden', function(newValue, oldValue) {
+          if ($scope.cm && oldValue === true && newValue !== oldValue) {
             bkUtils.fcall(function() {
-              $scope.cm.refresh();
+              $scope.afterShow();
```
Comment (tomca32) [inline core/src/main/web/app/mainapp/components/notebook/codecell-directive.js:105]: Do we need the call to afterShow here if the ngHideEx directive handles that automatically?

Comment (kpaharelau) [inline core/src/main/web/app/mainapp/components/notebook/codecell-directive.js:105]: Yes. ngHideEx is not working for hidden cells

File: core/src/main/web/app/mainapp/components/notebook/codecell-directive.js (hunk 1 above)
Comment (kpaharelau) [inline core/src/main/web/app/mainapp/components/notebook/codecell-directive.js]: good point - will be deleted

**Comment by scottdraves** [comment]
:+1:

**Pull request merged**